internal/k8s/job/job.go

```diff
@@ -0,0 +1,143 @@
+package job
+
+import (
+	"context"
+	"reflect"
+	"strings"
+	"time"
+
+	batchv1 "k8s.io/api/batch/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/vdaas/vald/internal/errors"
+	"github.com/vdaas/vald/internal/k8s"
+	"github.com/vdaas/vald/internal/log"
+)
+
+// JobWatcher is type alias for k8s.ResourceController.
+type JobWatcher k8s.ResourceController
+
+type reconciler struct {
+	ctx         context.Context
+	mgr         manager.Manager
+	name        string
+	namespance  string
+	onError     func(err error)
+	onReconcile func(jobList map[string][]Job)
+}
+
+// Job represents k8s job imformation
+// https://godoc.org/k8s.io/api/batch/v1#Job
+type Job struct {
```

Comment:
Why it is not a type alias for `k8s.io/api/batch/v1.Job`? it is better to use that one if this package is an internal package that is used generally in various applications.

Reply:
@kpango what do you think about this?

Reply:
In the Reconcile loop of the pod package, I had to calculate the CPU/memory requirements and limits of the pod, so I defined a specific type (not an alias for the type), but in the Job package I don't think we need to calculate it, so we can remove the Job struct and  add the type alias or use the directory reference of `k8s.io/api/batch/v1.Job`.

Reply:
FYI all of the current implementation of reconciler has the same problem/question 
e.g.
https://github.com/vdaas/vald/blob/master/internal/k8s/node/node.go#L91

Reply:
In node package and pod package we need to define specific structure type not type alias.